.drone.yml

```diff
@@ -159,6 +168,8 @@ steps:
     commands:
       - npm ci
       - npm run build
+    depends_on:
+      - promote to test check
```

Comment:
Were you able to verify that this step runs when we are promoting on main?
I'm curious because with the timecard one when the branch is actually main the 'check' step doesn't actually run at all.
E.g. https://drone-gh.acp.homeoffice.gov.uk/UKHomeOffice/callisto-timecard-restapi/563

And cos you're using a `depends_on` i'm wondering if this will even run if the 'check' step doesn't actually run when the promotion is valid.

Might be worth verifying if it works if you haven't already.

Reply:
Ah good idea, will test that out 👍

Reply:
From my extensive local testing (ahem) it doesn't seem to matter if the step isn't run. Which is what the [docs](https://docs.drone.io/pipeline/ssh/syntax/parallelism/) suggest too: `"Note that you can use conditional steps in your dependency graph. The scheduler automatically corrects the dependency graph for skipped steps." `

So I think we're good 🤞

Reply:
ah cool, looks good then 👍